index.js

```diff
@@ -612,10 +612,15 @@ class Hypercore extends EventEmitter {
     if (this.opened === false) await this.opening
     if (!isValidIndex(start) || !isValidIndex(end)) throw ASSERTION('has range is invalid')
 
-    if (end === start + 1) return this.state.bitfield.get(start)
+    let count = 0
```

Comment:
99/100 calls to has, is with the default end, ie one block, lets special case that as a fast path, ie , just this.state.storage.getBlock or whatever